eZ/Publish/Core/REST/Server/Input/Parser/Criterion/DateMetadata.php

```diff
@@ -6,26 +6,106 @@
  */
 namespace eZ\Publish\Core\REST\Server\Input\Parser\Criterion;
 
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion\Operator;
 use eZ\Publish\Core\REST\Common\Input\BaseParser;
 use eZ\Publish\Core\REST\Common\Input\ParsingDispatcher;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion\DateMetadata as DateMetadataCriterion;
+use eZ\Publish\Core\REST\Common\Exceptions;
 
 /**
  * Parser for ViewInput Criterion.
  */
 class DateMetadata extends BaseParser
 {
+    public const OPERATORS = [
+        'IN' => Operator::IN,
+        'EQ' => Operator::EQ,
+        'GT' => Operator::GT,
+        'GTE' => Operator::GTE,
+        'LT' => Operator::LT,
+        'LTE' => Operator::LTE,
+        'BETWEEN' => Operator::BETWEEN,
+    ];
+
+    /**
+     * DateMetadata target: modification date.
+     */
+    public const MODIFIED = 'modified';
+
+    /**
+     * DateMetadata target: creation date.
+     */
+    public const CREATED = 'created';
+
+    public const TARGETS = [
+        self::MODIFIED,
+        self::CREATED,
+    ];
```

Comment:
Personally, since we don't use those constants anywhere else and I'm unsure if we expect these to be used by customers, I'd keep those `private`, but that only my opinion and you can ignore it.

Reply:
If we don't need them anywhere else, then for sure we can mark then as `private` :wink:

Reply:
possibly I can change in the DateMetadata class to the public these consts and use them in this class (this will be done for the Ibex DXP 3+ version)

Reply:
Until they're not re-used, they should be private. From my point of view if something isn't used (even if predicted), then we shouldn't expose it.

After all, increasing visibility is a simple operation. Reducing it is not.

Reply:
Not to mention it seems that the actual owner of those constants should be the `DateMetadata` class, not the criterion, so they would eventually be moved around?

Reply:
I changed to use consts from the `DateMetadata` class